### DISCUSSION TITLE
Distinguishing behaviours from the sleep index and long periods of sitting/lying while awake (daytime)

### DISCUSSION THREAD

**Discussion by Suesmith12** [Ergo-Tools/ActiPASS#59] [Help and Q&A (processing, quality checking or statistics)]
In the attached QC visualisation, the sleep index lasts nearly all day on October 17th (this is also flagged in the warnings). In the attached sleep diary/log for the participant, they have specified that they were awake (ill with a migraine) during this time. 

![actipass QC check](https://github.com/Ergo-Tools/ActiPASS/assets/148974338/23c64fd8-3b79-44e4-829b-61570acb0679)
![sleep diary log](https://github.com/Ergo-Tools/ActiPASS/assets/148974338/b5bba828-1585-4e37-a024-7afaf99c1046)

I have been testing with the diary function (using our diary/logs as in the attached), but I’m not completely sure what this does? I did specify in the diary loaded with the data that this was time ‘Awake’, i.e., not to be classed with ‘Bed’ and I can see this in the visualisation - however, this time was still included as part of the sleep index for this participant. This would skew/affect their overall estimated avg sleep index time.

Our population is cancer patients where they may demonstrate long periods of day-time/awake sitting/lying and we would like to be able to class this as such (sedentary time sitting/lying outside of the sleep index), rather than it being part of sleep.

Please can you advise me as to if this would be possible to reclassify within ActiPASS?

**Comment by pasan-h** [comment]
> In the attached QC visualisation, the sleep index lasts nearly all day on October 17th (this is also flagged in the warnings). In the attached sleep diary/log for the participant, they have specified that they were awake (ill with a migraine) during this time.

This is a very important question. ActiPASS algorithms are developed and validated with healthy adults. It works with other populations with varying accuracy. One option is to reduce the behaviours it detect and make it more **robust**. We can do this (disabling/altering certain behaviour detection) step by step and try to understand what works. This will be a compromise (features against accuracy/robustness).

1. From the attached picture we see that the auto-bedtime-algorithm is over-sensitive for this person. And the sleep-algorithm (because it's based on purely movements) is also not smart enough to detect that the person is awake. One option is to chose a more robust (but somewhat less-sensitive and less smart) version of auto-bedtime-algorithm. To do this first we need to disable ProPASS mode and enable [advanced mode](https://github.com/Ergo-Tools/ActiPASS/wiki/ActiPASS-FAQ-and-troubleshooting#how-do-i-enable-actipass-advanced-mode-of-processing). Then we need to select "auto1" bedtime algorithm.   
    ![image](https://github.com/Ergo-Tools/ActiPASS/assets/26480941/f51a3a5b-0b29-4d3d-b143-2ac5b17db1cc)

2.  The second option is to disable sleep-detection completely. Then ActiPASS output will not contain sleep (or a sleep-interval), but if it can still flag periods as "possible" times-in-bed. To do this set sleep algorithm to "off"
   ![image](https://github.com/Ergo-Tools/ActiPASS/assets/26480941/9d662f8e-43f1-458c-af01-f449a2c4c14c)

3.  If auto detected times-in-bed are also bad (most probably they are for this population), and if you have a diary of their bedtimes, then we can use the diary for both times-in-bed and sleep (by definition, times-in-bed and sleep-interval are not the same, but for this population perhaps we can make them the same). To do this change settings as below.
    ![image](https://github.com/Ergo-Tools/ActiPASS/assets/26480941/5d71f025-fde9-483c-901e-5e2d7763bc82)


> 
> I have been testing with the diary function (using our diary/logs as in the attached), but I’m not completely sure what this does? I did specify in the diary loaded with the data that this was time ‘Awake’, i.e., not to be classed with ‘Bed’ and I can see this in the visualisation - however, this time was still included as part of the sleep index for this participant. This would skew/affect their overall estimated avg sleep index time.

Unfortunately there is no way to tell ActiPASS what are **awake times**. But we can change how it detects times-in-bed and sleep-interval. For an example by disabling the sleep-algorithm, there will be no sleep-interval and whole 24h will be considered awake. See above answers for various options we have related to times-in-bed and sleep.


> Our population is cancer patients where they may demonstrate long periods of day-time/awake sitting/lying and we would like to be able to class this as such (sedentary time sitting/lying outside of the sleep index), rather than it being part of sleep.
> 
> Please can you advise me as to if this would be possible to reclassify within ActiPASS?

We are thinking of making a more robust (but less features, less smart) mode of ActiPASS for populations such as yours. We need to identify what behaviours/activities which are essential, what we can compromise.  With populations such as yours, even behaviours such as star-walking and bicycling could be detected erroneously. This is because with limited mobility, ActiPASS automatic individual calibration can go wrong (such an individual calibration is needed for detecting certain behaviours)

Perhaps we can collaborate more towards making such a feature.

**Reply by Suesmith12** [reply]
Hi Pasan, 
Thank you so much for your quick response and for your suggestions. I will try your ideas with the data we have to see if there is any better approach to dealing with this population.
I will keep you updated.

**Comment by pasan-h** [comment]
> I have been testing with the diary function (using our diary/logs as in the attached), but I’m not completely sure what this does? I did specify in the diary loaded with the data that this was time ‘Awake’, i.e., not to be classed with ‘Bed’ and I can see this in the visualisation - however, this time was still included as part of the sleep index for this participant. This would skew/affect their overall estimated avg sleep index time.

Perhaps, my answer above is not very clear explaining how ActiPASS works with a times-in-bed/sleep diary. Under ProPASS mode, ActiPASS will use it's own times-in-bed and sleep algorithms. But if a diary is provided it merely shows diary times-in-bed in visualizations. Additionally it also compares automatic times-in-bed and sleep-interval with the diary. But it doesn't use the times defined in the diary in the actual output. In order for ActiPASS to consider the diary in it's output, we need to set either one or both **"Bedtime Alg"** and **"Sleep Alg"** options to **"diary"** like below.
![diary_bedtime_sleep](https://user-images.githubusercontent.com/26480941/278229206-5d71f025-fde9-483c-901e-5e2d7763bc82.png)
